src/logic/tokens/store/actions/fetchSafeTokens.js

```diff
@@ -78,8 +77,14 @@ const fetchSafeTokens = (safeAddress: string) => async (dispatch: ReduxDispatch<
     const updateActiveTokens = activeTokens.equals(alreadyActiveTokens) ? noFunc : update({ activeTokens })
     const updateBalances = balances.equals(safeBalances) ? noFunc : update({ balances })
     const updateEthBalance = ethBalance === currentEthBalance ? noFunc : update({ ethBalance })
-
-    const updateCurrencies = currencyList.equals(storedCurrencyBalances) ? noFunc : setCurrencyBalances(currencyList)
+    const storedCurrencyBalances =
+      currencyValues && currencyValues.get(safeAddress)
+        ? currencyValues.get(safeAddress).get('currencyBalances')
+        : undefined
+
+    const updateCurrencies = currencyList.equals(storedCurrencyBalances)
```

Comment:
isn't `equals` a pretty expensive operation? how often this action is called? I think we're doing something wrong if we have to check for an equality before dispatching any action

Reply:
This is to avoid unnecessary dispatch of actions maybe @fernandomg could add more about this :)

Reply:
Yes, it's an expensive operation.

But this is the only solution I came up with to avoid non-desired actions being triggered.